field/TinyLang/Field/Type.hs

```diff
@@ -0,0 +1,46 @@
+module TinyLang.Field.Type
+    ( Type(..)
+    , UniType
+    , pattern UniType
+    , pattern Bool
+    , pattern Field
+    , pattern Vector
+    ) where
+
+import           Data.Field
+import qualified TinyLang.Field.Uni         as U
+import           TinyLang.Field.Existential
+
+-- Types
+data Type uni f
+    = BuiltIn (Some (TypeIn uni f))
+    | TyFun   (Type uni f)          (Type uni f)
+
+type UniType = Type U.Uni
+
+pattern UniType :: U.Uni f a -> UniType f
+pattern UniType uni = BuiltIn (Some (TypeIn uni))
+
+deriving instance (TextField f) => Show (UniType f)
+deriving instance Eq  (UniType f)
+deriving instance Ord (UniType f)
+
+pattern Bool :: UniType f
+pattern Bool = UniType U.Bool
```

Comment:
Do we often need this "quick access"? I'd be fine with using `UniType Bool` explicitly instead of providing another term-level `Bool` name.

Reply:
So far we need it only in handful of places.  Will see later about it.